pkg/registry/core/endpoint/strategy.go

```diff
@@ -76,9 +77,30 @@ func (endpointsStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Ob
 
 // WarningsOnUpdate returns warnings for the given update.
 func (endpointsStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
-	return nil
+	return endpointsWarnings(obj.(*api.Endpoints))
 }
 
 func (endpointsStrategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+func endpointsWarnings(endpoints *api.Endpoints) []string {
+	// Don't bother checking for warnings if it looks like the request is coming from
+	// the EndpointController.
+	if endpoints.Annotations[api.EndpointsLastChangeTriggerTime] != "" {
```

Comment:
if we had this heuristic, why did we added the label in https://github.com/kubernetes/kubernetes/pull/130564

Reply:
People in the KEP wanted a more explicit indication than this heuristic. I need to update the PR now...